worker/src/RTC/SCTP/PacketItemBase.cpp

```diff
@@ -0,0 +1,183 @@
+#define MS_CLASS "RTC::SCTP::PacketItemBase"
+// #define MS_LOG_DEV_LEVEL 3
+
+#include "RTC/SCTP/PacketItemBase.hpp"
+#include "Logger.hpp"
+#include "MediaSoupErrors.hpp"
+#include <cstring> // std::memmove()
+#include <limits>  // std::numeric_limits()
+
+namespace RTC
+{
+	namespace SCTP
+	{
+		/* Class methods. */
+
+		bool PacketItemBase::IsPacketItemBase(
+		  const uint8_t* buffer, size_t bufferLength, uint16_t& itemLength, uint8_t& padding)
+		{
+			MS_TRACE();
+
+			if (bufferLength < PacketItemBase::PacketItemBaseHeaderLength)
+			{
+				MS_WARN_TAG(sctp, "no space for Header [bufferLength:%zu]", bufferLength);
+
+				return false;
+			}
+
+			itemLength = Utils::Byte::Get2Bytes(buffer, 2);
+
+			if (itemLength < PacketItemBase::PacketItemBaseHeaderLength)
+			{
+				MS_WARN_TAG(
+				  sctp,
+				  "Length field must have value greater or equal than %zu",
+				  PacketItemBase::PacketItemBaseHeaderLength);
+
+				return false;
+			}
+
+			// Item total length must be multiple of 4 bytes and must include padding
+			// bytes despite item Length field doesn't not include padding.
+			// NOTE: We must cast to size_t, otherwise a maximum item Length value of
+			// 65535 would generate a padded length of 0 bytes!
+			size_t paddedItemLength = Utils::Byte::PadTo4Bytes(size_t{ itemLength });
+
+			if (bufferLength < paddedItemLength)
+			{
+				MS_WARN_TAG(
+				  sctp,
+				  "no space for 4-byte padded announced Length [paddedItemLength:%zu, bufferLength:%zu]",
+				  paddedItemLength,
+				  bufferLength);
+
+				return false;
+			}
+
+			padding = paddedItemLength - itemLength;
+
+			return true;
+		}
+
+		/* Instance methods. */
+
+		PacketItemBase::PacketItemBase(uint8_t* buffer, size_t bufferLength)
+		  : Serializable(buffer, bufferLength)
+		{
+			MS_TRACE();
+		}
+
+		PacketItemBase::~PacketItemBase()
```

Comment:
Is this intended while development is ongoing? Otherwise why this rather than using a `=default`. Is it just for debugging?

Reply:
For debugging purposes and that saved my life many times.